packages/frontend-2/components/project/page/settings/general/block/Access.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <ProjectPageSettingsBlock
+    background
+    title="Access"
+    :icon="LockClosedIcon"
+    :disabled-message="
+      disabled && disabledMessage ? 'You must be a project owner' : undefined
+    "
+  >
+    <template #introduction>
+      <p>
+        Choose how you want to share your projects with others. Select the option that
+        best suits your needs:
+      </p>
+    </template>
+    <FormRadioGroup
+      v-model="selectedOption"
+      :options="radioOptions"
+      :disabled="disabled"
+      @update:model-value="emitUpdate"
+    />
+  </ProjectPageSettingsBlock>
+</template>
+
+<script setup lang="ts">
+import { LockClosedIcon, LinkIcon, GlobeAltIcon } from '@heroicons/vue/24/outline'
+import { FormRadioGroup } from '@speckle/ui-components'
+import { ProjectVisibility } from '~/lib/common/generated/gql/graphql'
+import { graphql } from '~~/lib/common/generated/gql'
+import type { ProjectPageSettingsGeneralBlockAccess_ProjectFragment } from '~~/lib/common/generated/gql/graphql'
+
+graphql(`
+  fragment ProjectPageSettingsGeneralBlockAccess_Project on Project {
+    id
+    visibility
+  }
+`)
+
+const props = defineProps<{
+  project: ProjectPageSettingsGeneralBlockAccess_ProjectFragment
+  disabled?: boolean
+  disabledMessage?: string
+}>()
+
+const emit = defineEmits(['update-visibility'])
+
+const selectedOption = ref(props.project.visibility ?? ProjectVisibility.Private)
```

Comment:
Not a big deal, but `||` is in most cases better than `??`, because surely you'd want it to fall back to that value even if in the case of a bug `props.project.visibility` is `false` or `''` (empty string) or `0` or any other kind of falsy value